public/app/features/dashboard-scene/settings/AnnotationsEditView.tsx

```diff
@@ -1,29 +1,89 @@
 import React from 'react';
 
-import { PageLayoutType } from '@grafana/data';
-import { SceneComponentProps, SceneObjectBase } from '@grafana/scenes';
+import { AnnotationQuery, DataTopic, PageLayoutType } from '@grafana/data';
+import { locationService } from '@grafana/runtime';
+import { SceneComponentProps, SceneDataLayerProvider, SceneObjectBase, sceneGraph } from '@grafana/scenes';
 import { Page } from 'app/core/components/Page/Page';
 
+import { DashboardScene } from '../scene/DashboardScene';
 import { NavToolbarActions } from '../scene/NavToolbarActions';
+import { dataLayersToAnnotations } from '../serialization/dataLayersToAnnotations';
 import { getDashboardSceneFor } from '../utils/utils';
 
+import { AnnotationSettingsList } from './annotations';
 import { DashboardEditView, DashboardEditViewState, useDashboardEditPageNav } from './utils';
 
-export interface AnnotationsEditViewState extends DashboardEditViewState {}
+export interface AnnotationsEditViewState extends DashboardEditViewState {
+  editIndex?: number | null;
+}
 
 export class AnnotationsEditView extends SceneObjectBase<AnnotationsEditViewState> implements DashboardEditView {
   public getUrlKey(): string {
     return 'annotations';
   }
 
+  constructor(state: AnnotationsEditViewState) {
+    super({
+      ...state,
+      editIndex: null,
+    });
+
+    this.addActivationHandler(() => {
+      this.setState({ editIndex: this.getEditIndex() });
+    });
+  }
+
+  private get _dashboard(): DashboardScene {
+    return getDashboardSceneFor(this);
+  }
+
+  private get _dataLayers(): SceneDataLayerProvider[] {
+    return sceneGraph.getDataLayers(this._dashboard);
+  }
+
+  private getEditIndex() {
```

Comment:
why this is not a `private get`? Just to be consistent :)

Reply:
this ain't needed at all, so I suggest skipping it.

Reply:
I removed all `editIndex` related logic and used `_urlSync`